packages/odyssey-react-mui/src/Badge.tsx

```diff
@@ -0,0 +1,52 @@
+/*!
+ * Copyright (c) 2023-present, Okta, Inc. and/or its affiliates. All rights reserved.
+ * The Okta software accompanied by this notice is provided pursuant to the Apache License, Version 2.0 (the "License.")
+ *
+ * You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *
+ * See the License for the specific language governing permissions and limitations under the License.
+ */
+
+import { ReactNode, memo } from "react";
+import { Badge as MuiBadge, BadgeProps as MuiBadgeProps } from "@mui/material";
+import type { SeleniumProps } from "./SeleniumProps";
+
+export const badgeTypeValues = ["default", "primary", "error"] as const;
+export const badgeVariantValues = ["dot", "standard"] as const;
+
+export type BadgeProps = {
+  children?: ReactNode;
+  badgeContent?: number;
+  badgeContentMax?: MuiBadgeProps["max"];
+  variant?: (typeof badgeVariantValues)[number];
+  type?: (typeof badgeTypeValues)[number];
+} & SeleniumProps;
+
+const Badge = ({
+  children,
+  badgeContent,
+  badgeContentMax = 999,
+  variant = "standard",
+  type = "default",
+}: BadgeProps) => {
+  const threeDigitLimitedMax = badgeContentMax > 999 ? 999 : badgeContentMax;
```

Comment:
What's with the `999`?

Reply:
@KevinGhadyani-Okta We wanted to ensure that the content never goes past 3 chars. So, if the consumer passed in a max of 2000, the component would still limit it to 3 chars.

Reply:
If it goes over 999, would it count as 4 characters to have it show 999+? Just wondering. Support 3 characters because it saves us from comma/period internationalization issues with 1000 etc.

Reply:
That's a great point!

Reply:
I hadn't thought about the complications with comma/period. @benlister-okta just wanted the max number to be 3 chars long. We're fine with 3 chars and `+`.